### DISCUSSION TITLE
Technical Lead Team rolling issues, shoutouts, and meeting deciders - Part 1, to be discussed at TLT Meeting 12 Jan 2023

### DISCUSSION THREAD

**Discussion by valeriupredoi** [ESMValGroup/Community#68] [Meetings]
Followup item from #62 - @ESMValGroup/technical-lead-development-team please post your new ideas for discussion items here, we will try and organize a TLT call in the first week of January 2023 (sometime between 9 and 13 Jan, I will bug people with a Doodle after NY day). In the meantime, @remi-kazeroni and myself will put together the minutes from yesterday's TLT call and add a link [here](https://github.com/ESMValGroup/Community/blob/main/Tech%20Lead/Minutes/20221213.md), so things get organized nicely; also, have a look at #62 and please move any points from there that you feel still need touching up on :beer: :christmas_tree: 

**EDIT** with the [minutes](https://github.com/ESMValGroup/Community/blob/main/Tech%20Lead/Minutes/20221213.md) from TLT 13 December 2022: note that points 6, 7 and 8 are TBD, maybe we can touch upon them before the next release, ie at the January call.

**Comment by zklaus** [comment]
Should we open the Slack to the general (ESMValTool) public?

**Reply by valeriupredoi** [reply]
I like the idea! But what do you mean by *general (ESMValTool) public* - the developers (all) or just other teams (science, outreach etc)? Or you have this in mind as another really open platform for general support, like jasmin_general for anyone with Slack and using ESMValTool?

**Reply by remi-kazeroni** [reply]
@bouweandela, @zklaus: would be positive about that
@valeriupredoi, @remi-kazeroni: have some reservations that users would stop using github to report issues.  
@Saskia: experience from other projects not very positive
Opening access: to be discussed with the other teams. As a first step we could open access to other teams and structure our Slack a bit more.

**Comment by remi-kazeroni** [comment]
#### Approaches to run all recipes (during releases)

Follow up on https://github.com/ESMValGroup/Community/discussions/62#discussioncomment-4119346

**Comment by remi-kazeroni** [comment]
#### Release procedure: documentation

[Originally posted by @valeriupredoi]

release procedure for ESMValTool is involving and not well documented - I'd like to discuss https://github.com/ESMValGroup/ESMValTool/issues/2895 and also the spread of release workload - I think we can safely say that 2 people (release managers) would be needed in the eventuality that RTW doesn't come online before the next release (even so, it might be experimental, and a full run of all the recipes might still be needed one last release). I would also want to touch upon https://github.com/ESMValGroup/ESMValTool/issues/2895 (also mentioned in my previous comment) and specifically data dirs on The Big Levante

@bouweandela adds to this: This would also be a good and related discussion topic: https://github.com/ESMValGroup/ESMValTool/pull/2883

**Reply by bouweandela** [reply]
From the [minutes of the previous meeting](https://github.com/ESMValGroup/Community/blob/main/Tech%20Lead/Minutes/20221213.md):

> Documentation @valeriupredoi will update the release documentation early January, previous release managers can then take a look

Has this happened?

**Comment by remi-kazeroni** [comment]
#### Administration of the VM

Follow up on https://github.com/ESMValGroup/Community/discussions/62#discussioncomment-4148563

**Comment by remi-kazeroni** [comment]
#### Upcoming release v2.8

Shall we have a quick look at the [Core milestone v2.8](https://github.com/ESMValGroup/ESMValCore/milestone/14)? We could focus only on the few "biggest" PRs (e.g. is-enes3 related ones) and check what is the timeline to complete the work, review, test and merge. I'm just wondering if starting the release process a couple of weeks later would help to have everything we need and promised properly integrated.

**Comment by remi-kazeroni** [comment]
## Agenda - Jan 12, 2023

1. Action items for previous meeting: 
  - common Gmail address to share credentials for services used to develop ESMValTool. See [here](https://github.com/ESMValGroup/Community/discussions/62#discussioncomment-4119360) (fuuuu... forgot again, will do TODAY!)
2. [Opening the Slack to the general (ESMValTool) public](https://github.com/ESMValGroup/Community/discussions/68#discussioncomment-4399940)
3. [Upcoming release v2.8 - status of major PRs](https://github.com/ESMValGroup/Community/discussions/68#discussioncomment-4665079)
4. [Release procedure: documentation](https://github.com/ESMValGroup/Community/discussions/68#discussioncomment-4665011)
5. [Approaches to run all recipes (during releases)](https://github.com/ESMValGroup/Community/discussions/68#discussioncomment-4665011)
6. [Administration of the VM](https://github.com/ESMValGroup/Community/discussions/68#discussioncomment-4665018)
x134. Use [`.conda`](https://conda.io/projects/conda/en/latest/user-guide/concepts/packages.html#conda-file-format) package fileformat output?

This list contains links to all suggested topics for this meeting. We can reorder the list at the beginning of the meeting.

**Comment by valeriupredoi** [comment]
@remi-kazeroni @valeriupredoi for minutes:

3. Release 2.8: milestone 2.8 list examined, aiming at the big/important PRs:
   - @bouweandela
     - a bit slower since review is at a premium - ESMValGroup/ESMvalCore#1609 is close to be merged (dataset versioning is important);
     - ESMValGroup/ESMvalCore#1877 is important but needs an overhaul of recipes handling -> maybe include it after release? Downside: IS-ENES3 feature will not be included (@remi-kazeroni stresses the fact that it's an IS-ENES3 milestone release);
     - @bouweandela suggests @remi-kezeroni decides if this is really needed for the IS-ENES3-themed release;
     - @valeriupredoi possible to delay the release so we can get that bug PR in, and test only once;
     - @zklaus may be a problem once the IS-ENES3 money dried up? @bouweandela says his money is on low, so big feature work should be done by April
     - @bouweandela sets the roadmap: ESMValGroup/ESMvalCore#1877 into `main` then `main` back into ESMValGroup/ESMvalCore#1609 that will be split in little pullets
     - @bouweandela - testing should (normally) be done only for the release
   - going through PRs for 2.8 milestone list, and discussing which one is good to be in the release or not;
     - @schlunma ESMValGroup/ESMvalCore#1664 will be reviewed by @bsolino and @bouweandela 
   - going through the 2.8 milestone issues also;
     - one thing is the CMOR obs4mips tables are ye olde https://github.com/ESMValGroup/ESMValCore/issues/1890
     - don't pin iris
     
**Delay release by 2 wks**
